profiles/common/modules/custom/nexteuropa_token/src/Entity/ViewModeTokenHandler.php

```diff
@@ -44,59 +44,45 @@ public function hookTokens($type, $tokens, array $data = array(), array $options
     if ($this->isValidTokenType($type)) {
       foreach ($tokens as $name => $original) {
         if ($this->isValidToken($original)) {
-          $render = array();
-          $entity_id = $this->getEntityIdFromToken($original);
-          $view_mode = $this->getViewModeFromToken($original);
-
-          switch ($type) {
-            case 'node':
-              if ($node = node_load($entity_id)) {
-                if ($this->canViewNode($node)) {
-                  $render = node_view($node, $view_mode);
-                }
+          $entity_view_mode_type = $this->getViewModeTypeHandlers();
+
+          // By default let this be the token, not converted.
+          $render = array(
+            '#markup' => $original,
+          );
+
+          if (isset($entity_view_mode_type[$type])) {
+            // Initialize the object.
+            $entity_view_mode_type = $this->getEntityViewModeType($type, array(
+              'entity id' => $this->getEntityIdFromToken($original),
+              'view mode' => $this->getViewModeFromToken($original),
+            )
+            );
+
+            // Before trying to render, check if the view mode is available.
+            if ($entity_view_mode_type->isValidViewMode()) {
+              if ($render_tmp = $entity_view_mode_type->entityView()) {
+                // If the render is successful, overwrite the default
+                // variable content.
+                $render = $render_tmp;
               }
-              else {
-                $this->watchdogTokenNotFound($data, $original);
-              }
-              break;
-
-            case 'term':
-              if ($term = taxonomy_term_load($entity_id)) {
-                $render = taxonomy_term_view($term, $view_mode);
-              }
-              else {
-                $this->watchdogTokenNotFound($data, $original);
-              }
-              break;
-
-            case 'user':
-              if ($account = user_load($entity_id)) {
-                if (user_access('access user profiles')) {
-                  $render = user_view($account, $view_mode);
-                }
-              }
-              else {
-                $this->watchdogTokenNotFound($data, $original);
-              }
-              break;
+            }
+          }
+          else {
+            // If no plugin handler is found for the entity
+            // use the default entity API to render it.
+            $entities = entity_load($type, array($this->getEntityIdFromToken($original)));
```

Comment:
Maybe we just don't render it if no handler is found? This could allow administrators to "hack" into the rendering process by just guessing the right entity type machine name, right?